source/GameInterface/DynamicSync/Builders/DynamicSyncFieldBuilder.cs

```diff
@@ -0,0 +1,118 @@
+﻿using GameInterface.DynamicSync.Templates;
+using GameInterface.Services.ObjectManager;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace GameInterface.DynamicSync.Builders
+{
+    public class DynamicSyncFieldBuilder
+    {
+        private readonly IObjectManager objectManager;
+
+        public DynamicSyncFieldBuilder(IObjectManager objectManager)
+        {
+            this.objectManager = objectManager;
+        }
+        public string GetTranspiler(FieldInfo fieldInfo)
+        {
+            var template = TemplateParser.Parse("Patches.FieldSetTranspilerTemplate",
+                new
+                {
+                    MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                    MemberName = fieldInfo.Name,
+                    MemberType = fieldInfo.FieldType.Name
+                });
+            // TODO: Register Intercept for test usages
+            return template;
+        }
+
+        public IEnumerable<string> GetMessages(FieldInfo fieldInfo)
+        {
+            string localMessage = TemplateParser.Parse("Messages.LocalSetMessageTemplate",
+                new
+                {
+                    MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                    MemberName = fieldInfo.Name,
+                    MemberType = fieldInfo.FieldType.Name,
+                    Libraries = new List<string>
+                    {
+                        fieldInfo.DeclaringType.Namespace,
+                        fieldInfo.FieldType.Namespace
+                    }
+                });
+            string networkMessage;
+            if (objectManager.IsTypeManaged(fieldInfo.FieldType))
+            {
+                networkMessage = TemplateParser.Parse("Messages.NetworkSetReferenceMessageTemplate",
+                    new
+                    {
+                        MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                        MemberName = fieldInfo.Name,
+                        MemberType = fieldInfo.FieldType.Name,
+                        Libraries = new List<string>
+                        {
+                        fieldInfo.DeclaringType.Namespace,
+                        fieldInfo.FieldType.Namespace
+                        }
+                    });
+            }
+            else
```

Comment:
This should check if the type is serializable via protobuf, if not throw exception

https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/AutoSync/Properties/PropertyPrefixCreator.cs#L248